Controllers/HomeController.cs

```diff
@@ -24,11 +37,45 @@ public IActionResult About()
             return View();
         }
 
-        public IActionResult Contact()
+        public async Task<IActionResult> Contact()
         {
-            ViewData["Message"] = "Your contact page.";
+            string[] scopes = new string[] { "user.read" };
+            try
 
-            return View();
+            {
+                string accessToken = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(HttpContext, User, scopes);
+                dynamic me = await CallGraphApiOnBehalfOfUser(accessToken);
+
+                ViewData["Me"] = me;
+                return View();
+            }
+            catch (MsalException)
+            {
+                var redirectUrl = Url.Action(nameof(HomeController.Contact), "Home");
+                return Challenge(
+                    new AuthenticationProperties { RedirectUri = redirectUrl, IsPersistent = true },
+                    OpenIdConnectDefaults.AuthenticationScheme);
+            }
+        }
+
+        private static async Task<dynamic> CallGraphApiOnBehalfOfUser(string accessToken)
+        {
+            //
+            // Call the Graph API and retrieve the user's profile.
+            //
+            HttpClient client = new HttpClient();
```

Comment:
There are some locations in the code where 'var' is used and other locations where it isn't. I think at the very least it would be good to be consistent about it.

Reply:
I've changed this to use 'var'.